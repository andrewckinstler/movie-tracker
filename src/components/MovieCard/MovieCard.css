.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px;
  max-width: 453px;
  padding: 3px;
  background-color: #465C69;
  border-radius: 4px;
  text-align: center;
  
}

h3 {
  flex: 1;
  color: #000;
  text-decoration: none;
  overflow-wrap: break-word;
}

button {
  margin: 10px 0;
  height: 20px;
  width: 120px;
  font-size: .75em;
  border: none;
  border-radius: 4px;
  background-color: #A2AEBB;
  color: #FFF;
}