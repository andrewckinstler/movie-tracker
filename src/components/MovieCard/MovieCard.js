import React from 'react';
import { Link } from 'react-router-dom';
import { addToWatchList } from '../../actions/index'
import { connect } from 'react-redux'
import './MovieCard.css'

export const MovieCard = ({ title, poster_path, id, addToWatchList, movies }) => {
  const image = `https://image.tmdb.org/t/p/w300/${poster_path}`

  const movie = movies.find(movie => movie.id === id)

  return (
    <div className='card'>
      <Link to={`movie/${id}`}>
        <article>
          <img src={image} />
          <div className='title-container'>
            
          </div>
        </article>
      </Link>
      <h3>{title}</h3>
      <div>
      <button onClick={() => addToWatchList(movie)}>Add to your watchlist</button>
      </div>
    </div>
  )
}

export const mapStateToProps = state => ({
  movies: state.movies
})

export const mapDispatchToProps = dispatch => ({
  addToWatchList: toAdd => dispatch( addToWatchList(toAdd) )
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieCard)
