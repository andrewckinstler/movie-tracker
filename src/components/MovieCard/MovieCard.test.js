import React from 'react';
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard';
import { shallow } from 'enzyme';
import { addToWatchList } from '../../actions/index';

describe('MovieCard', () => {
  let wrapper, mockMovie, movies;
  beforeEach(() => {
    movies = [
      {
          "genre_ids": [
              27,
              9648,
              878,
              53
          ],
          "id": 555974,
          "title": "The Invisible Man",
          "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
          "release_date": "2020-02-26"
      },
      {
          "genre_ids": [
              27,
              9648,
              53
          ],
          "id": 570670,
          "title": "Brahms: The Boy II",
          "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
          "release_date": "2020-02-20"
      }
    ]
    mockMovie = {
      "genre_ids": [
          27,
          9648,
          878,
          53
      ],
      "id": 555974,
      "poster_path": "/4U7hpTK0XTQBKT5X60bKmJd05ha.jpg",
      "title": "The Invisible Man",
      "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
      "release_date": "2020-02-26"
    }
    wrapper = shallow(<MovieCard movies={movies} {...mockMovie} key={mockMovie.id} />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return an object with an array of movies', () => {
  
      const mockState = {
        movies: [
          {
              "genre_ids": [
                  27,
                  9648,
                  878,
                  53
              ],
              "id": 555974,
              "title": "The Invisible Man",
              "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
              "release_date": "2020-02-26"
          },
          {
              "genre_ids": [
                  27,
                  9648,
                  53
              ],
              "id": 570670,
              "title": "Brahms: The Boy II",
              "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
              "release_date": "2020-02-20"
          }
        ]
      }
  
      const expected = {
        movies: [
          {
              "genre_ids": [
                  27,
                  9648,
                  878,
                  53
              ],
              "id": 555974,
              "title": "The Invisible Man",
              "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
              "release_date": "2020-02-26"
          },
          {
              "genre_ids": [
                  27,
                  9648,
                  53
              ],
              "id": 570670,
              "title": "Brahms: The Boy II",
              "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
              "release_date": "2020-02-20"
          }
        ]
      }
  
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with an addMovies action when addMovies is called', () => {
      const mockDispatch = jest.fn();
      const movie = {
            "genre_ids": [
                27,
                9648,
                878,
                53
            ],
            "title": "The Invisible Man",
            "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
            "release_date": "2020-02-26"
        }
      const action = addToWatchList(movie)
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.addToWatchList(movie);
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })
  })
})