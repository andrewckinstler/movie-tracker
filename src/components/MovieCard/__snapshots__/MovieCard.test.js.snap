// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match a snapshot 1`] = `
<div
  className="card"
>
  <Link
    to="movie/555974"
  >
    <article>
      <img
        src="https://image.tmdb.org/t/p/w300//4U7hpTK0XTQBKT5X60bKmJd05ha.jpg"
      />
      <h3>
        The Invisible Man
      </h3>
    </article>
  </Link>
  <div>
    <button
      onClick={[Function]}
    >
      Add to your watchlist
    </button>
  </div>
</div>
`;
